packages/design-system/src/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { useArgs } from "@storybook/client-api";
+
+import {
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+} from "./Modal";
+import { Button } from "../Button";
+
+export default {
+  title: "Design System/Modal",
+  component: Modal,
+  argTypes: {
+    open: {
+      control: {
+        type: "boolean",
+      },
+      description:
+        "Open state of the modal. Make sure onClose is passed in ModalHeader.",
+      table: {
+        type: {
+          summary: "boolean",
+        },
+        defaultValue: {
+          summary: "false",
+        },
+      },
+    },
+    onOpenChange: {
+      action: "onOpenChange",
+      description: "Callback for when the modal is opened or closed",
+      table: {
+        type: {
+          summary: "(open: boolean) => void",
+        },
+        defaultValue: {
+          summary: "undefined",
+        },
+      },
+    },
+    children: {
+      description: "Content of the modal",
+      table: {
+        type: {
+          summary: "React.ReactNode",
+        },
+        defaultValue: {
+          summary: "undefined",
+        },
+      },
+    },
+  },
+  subcomponents: {
+    ModalContent,
+    ModalHeader,
+    ModalBody,
+    ModalFooter,
+  },
+} as ComponentMeta<typeof Modal>;
+
+// eslint-disable-next-line react/function-component-definition
+const Template: ComponentStory<typeof Modal> = (args) => {
+  const [{ open }, updateArgs] = useArgs();
+  const handleClose = () => updateArgs({ open: !open });
+
+  return (
+    <>
+      <Button onClick={handleClose}>Open Modal</Button>
+      <Modal {...args}>
+        <ModalContent>
+          <ModalHeader onClose={handleClose}>Modal Header</ModalHeader>
+          <ModalBody>Test body</ModalBody>
+          <ModalFooter>
+            <Button kind="secondary" size="md">
```

Comment:
We were going to hard code this in the footer to prevent bad patterns no?

Reply:
We cannot hard code this in footer since we are not sure on how many or which type of buttons they want. If Vasanth can confirm the pattern then I can make that change.